Sources/libgir2swift/utilities/planning.swift

```diff
@@ -0,0 +1,203 @@
+        case prerequisities
+    }
+
+    let version: UInt
+    let girName: String
+    let pkgConfig: String
+    let prerequisities: [Prerequisity]?
+}
+
+struct GirPackageMetadata {
+    struct Dependency {
+        let name: String
+        let version: String
+
+        var girName: String { name + "-" + version }
+    }
+
+    let pkgName: String
+    let dependency: [Dependency]
+}
+
+struct Plan {
+
+    enum Error: Swift.Error {
+        case girNotFound(named: String)
+        case girParsingFailed
+    }
+
+    let girFileToGenerate: URL
+    let girFilesToPreload: [URL]
+    let pkgConfigName: String
+
+    init(using manifestUrl: URL) throws {
+        // From some reason, Data would not provide correct result
+        let data = try String.init(contentsOfFile: manifestUrl.path)
+        let manifest = try YAMLDecoder().decode(Manifest.self, from: data)
+        
+        let optGirPath = try Plan.searchForGir(
+            named: manifest.girName, 
+            pkgConfig: [manifest.pkgConfig]
+        )
+        guard let girPath = optGirPath else {
+            throw Error.girNotFound(named: manifest.girName + ".gir" )
+        } 
+
+        self.girFileToGenerate = girPath
+        self.girFilesToPreload = try Plan.loadPrerequisities(from: girPath, pkgConfig: manifest.pkgConfig, prerequisities: manifest.prerequisities)
+        self.pkgConfigName = manifest.pkgConfig
+    }
+
+    private static func searchForGir(named name: String, pkgConfig: Set<String>) throws -> URL? {
+        let defaultPaths = ["/opt/homebrew/share/gir-1.0", "/usr/local/share/gir-1.0", "/usr/share/gir-1.0"].map { URL.init(fileURLWithPath: $0, isDirectory: false) }
+
+        #if os(macOS)
+        let homebrewRelativeGirLocation = "../share/gir-1.0/"
+        let homebrewPaths = pkgConfig.compactMap { pkgName -> URL? in 
+            let libDir = try? executeAndWait("pkg-config", arguments: ["--variable=libdir", pkgName])
+            return libDir.flatMap { 
+                let libDirUrl = URL(fileURLWithPath: $0, isDirectory: true)
+                return URL(string: homebrewRelativeGirLocation, relativeTo: libDirUrl)
+            }
+        }
+
+        let searchPaths = homebrewPaths + defaultPaths
+        #else
+        let searchPaths = defaultPaths
+        #endif
+
+
+        return searchPaths
+            .map { $0.appendingPathComponent(name).appendingPathExtension("gir") }
+            .first { 
+                FileManager.default.fileExists(atPath: $0.path) 
+            }
+    }
+
+    private static func loadPrerequisities(from gir: URL, pkgConfig: String, prerequisities: [Prerequisity]?) throws -> [URL] {
+        var explored: [String: URL] = [ gir.deletingLastPathComponent().lastPathComponent : gir ]
+        var toExplore: Set<String> = Set(try parsePackageInfo(for: gir).dependency.map(\.girName) + (prerequisities?.map(\.girName) ?? []))
+
+        var pkgConfigCandidates: Set<String> = []
+        #if !os(macOS)
```

Comment:
shouldn't this be `#if os(macOS)`?

Reply:
Sure! 

Honestly, I should just go to bed right now :D so I won’t create some more things like this :D

(_Since I work on Linux, I needed this macro inverted. Otherwise, the sourcekit would ignore it. Thanks :+1:_)